File: mybudget-backend/src/main/java/com/Project/MyBudget/services/income/IncomeService.java
package com.Project.MyBudget.services.income;

import com.Project.MyBudget.dto.IncomeDTO;
import com.Project.MyBudget.entity.Income;

import java.util.List;

public interface IncomeService {

    Income postIncome(IncomeDTO incomeDTO);

    List<IncomeDTO> getAllIncomes();

    Income updateIncome(Long id, IncomeDTO incomeDTO);

    IncomeDTO getIncomeById(Long id);

    void deleteIncome(Long id);
}

File: mybudget-frontend/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  private baseUrl = `${environment.apiUrl}api/income`;

  constructor(private http: HttpClient) { }

  postIncome(incomeDTO: any): Observable<any> {
    return this.http.post(this.baseUrl, incomeDTO);
  }

  getAllIncomes(): Observable<any> {
    return this.http.get(`${this.baseUrl}/all`);
  }

  getIncomeById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  updateIncome(id: number, incomeDTO: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, incomeDTO);
  }

  deleteIncome(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }
}
